File: 251002/Korti/publisher/src/entities/Note.ts
import {
  Column,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
} from 'typeorm';
import { Article } from './Article';
@Entity({ name: 'tbl_note' })
export class Note {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ nullable: false })
  articleId: number;

  @Column({ nullable: false })
  content: string;

  @ManyToOne(() => Article, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'articleId' })
  article: Article;
}

File: 251002/Samoilenko/publisher/src/main/java/com/example/discussion/model/Note.java
package com.example.discussion.model;

import jakarta.persistence.*;

@Entity
@Table(name = "tbl_note")
public class Note {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "content", nullable = false, length = 2048)
    private String content;

    @ManyToOne
    @JoinColumn(name = "news_id", nullable = false)
    private News news;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public News getNews() {
        return news;
    }

    public void setNews(News news) {
        this.news = news;
    }
}